From c4463e6d54a1335daa59c9c32bd1d89225a553cf Mon Sep 17 00:00:00 2001
From: yangson1 <103113662+yangson1@users.noreply.github.com>
Date: Fri, 15 Apr 2022 09:41:22 -0700
Subject: Add support to td gateway in gcp_envoy_bootstrap.json (#38393)

* Add support to td gateway name in gcp_envoy_bootstrap.json

* Update gcp_envoy_bootstrap.json
---
 tools/packaging/common/gcp_envoy_bootstrap.json | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/packaging/common/gcp_envoy_bootstrap.json b/tools/packaging/common/gcp_envoy_bootstrap.json
index 9a4ae9c284..507e1e9fc9 100644
--- a/tools/packaging/common/gcp_envoy_bootstrap.json
+++ b/tools/packaging/common/gcp_envoy_bootstrap.json
@@ -2,6 +2,8 @@
   "node": {
     {{ if (index .metadata.Labels "td.networking.gke.io/mesh-name") }}
       "id": "projects/{{ .gcp_project_number }}/networks/mesh:{{ (index .metadata.Labels "td.networking.gke.io/mesh-name") }}/nodes/{{ .nodeID}}",
+    {{ else if (index .metadata.Labels "td.networking.gke.io/gateway-name") }}
+      "id": "projects/{{ .gcp_project_number }}/networks/scope:{{ (index .metadata.Labels "td.networking.gke.io/gateway-name") }}/nodes/{{ .nodeID}}",
     {{ else }}
       "id": "projects/{{ .gcp_project_number }}/networks/default/nodes/{{ .nodeID}}",
     {{ end }}
-- 
2.35.3

